examples/platforms/nrf52840/diag.c

```diff
@@ -239,6 +241,87 @@ static void processTransmit(otInstance *aInstance, int argc, char *argv[], char
     appendErrorResult(error, aOutput, aOutputMaxLen);
 }
 
+static void processGpio(otInstance *aInstance, int argc, char *argv[],
+                        char *aOutput, size_t aOutputMaxLen)
+{
+    OT_UNUSED_VARIABLE(aInstance);
+    OT_UNUSED_VARIABLE(argv);
+
+    long pinnum;
+    otError error = OT_ERROR_NONE;
+
+    otEXPECT_ACTION(otPlatDiagModeGet(), error = OT_ERROR_INVALID_STATE);
+
+    if (argc == 1)
+    {
+        uint32_t value;
+
+        error = parseLong(argv[0], &pinnum);
+        otEXPECT(error == OT_ERROR_NONE);
+
+        value = nrf_gpio_pin_read(pinnum);
+
+        snprintf(aOutput, aOutputMaxLen, "gpio %d = %d\r\n",
+                 (int)pinnum, (int)value);
```

Comment:
nit: Is it possible to avoid conversions by changing the format string to "gpio %ld = %lu\r\n"? Same to other similar places.

Reply:
I could do that. Though for embedded platforms, I think casting to a smaller type and triggering the simpler %d parser is more efficient.